presto-druid/src/main/java/io/prestosql/plugin/druid/DruidJdbcClient.java

```diff
@@ -0,0 +1,365 @@
+        return metadata.getTables(DRUID_CATALOG,
+                DRUID_SCHEMA,
+                tableName.orElse(null),
+                null);
+    }
+
+    @Override
+    public Optional<ColumnMapping> toPrestoType(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        switch (typeHandle.getJdbcType()) {
+            case Types.VARCHAR:
+                int columnSize = typeHandle.getColumnSize();
+                if (columnSize > VarcharType.MAX_LENGTH || columnSize == -1) {
+                    return Optional.of(varcharColumnMapping(createUnboundedVarcharType()));
+                }
+                return Optional.of(varcharColumnMapping(createVarcharType(columnSize)));
+            default:
+                return super.toPrestoType(session, connection, typeHandle);
+        }
+    }
+
+    /**
+     * Druid doesn't like table names to be qualified with catalog names in the SQL query.
+     * Hence, overriding this method to pass catalog as an empty string.
+     */
+    @Override
+    public PreparedStatement buildSql(ConnectorSession session, Connection connection, JdbcSplit split, JdbcTableHandle table, List<JdbcColumnHandle> columns)
+            throws SQLException
+    {
+        String schemaName = table.getSchemaName();
+        checkArgument(schemaName.isEmpty() || schemaName.equals("druid"), "Only \"druid\" schema is supported");
+        return new QueryBuilder(identifierQuote)
+                .buildSql(
+                        this,
+                        session,
+                        connection,
+                        "",
+                        schemaName,
+                        table.getTableName(),
+                        columns,
+                        table.getConstraint(),
+                        split.getAdditionalPredicate(),
+                        tryApplyLimit(table.getLimit()));
+    }
+
+    /**
+     * Overriden to filter out tables that don't match the @param tableHandle
+     * <p>
+     * See {@link #getColumns(JdbcTableHandle, DatabaseMetaData)}
+     */
+    @Override
+    public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHandle tableHandle)
+    {
+        try (Connection connection = connectionFactory.openConnection(JdbcIdentity.from(session));
+                ResultSet resultSet = getColumns(tableHandle, connection.getMetaData())) {
+            int allColumns = 0;
+            List<JdbcColumnHandle> columns = new ArrayList<>();
+            while (resultSet.next()) {
+                // skip if table doesn't match expected
+                if (!(Objects.equals(tableHandle.getCatalogName(), resultSet.getString("TABLE_CAT"))
+                        && Objects.equals(tableHandle.getSchemaName(), resultSet.getString("TABLE_SCHEM"))
+                        && Objects.equals(tableHandle.getTableName(), resultSet.getString("TABLE_NAME")))) {
+                    continue;
```

Comment:
Is this the only thing being added here?

If we added this in BaseJdbcClient, would it allow us not to override this (lengthy) method?

Reply:
Yes, I wouldn't need to override the method then. Wasn't too sure if it would be ok to move the check in the base class's method. Will move the check in the base class and hopefully tests for other connectors will still pass (they do pass for the Druid connector).